<launch>

<group ns="u2dar_receiver">

<arg name="i2c_dev" default="/dev/i2c-1" />
<arg name="spi_dev" default="/dev/spidev0.0" />
<arg name="alsa_dev" default="hw:0" />
<arg name="pga_gain" default="157.0" />
<arg name="pga_offset" default="0.0" />
<arg name="swcf_a_f0" default="20000.0" />
<arg name="swcf_a_q" default="1.0" />
<arg name="swcf_a_mode" default="1" />
<arg name="swcf_b_f0" default="20000.0" />
<arg name="swcf_b_q" default="1.0" />
<arg name="swcf_b_mode" default="1" />

<!-- launch-prefix="debug_roslaunch" -->

<node name="recorder_node" pkg="u2dar_receiver" type="recorder_node" output="screen" >
    <param name="i2c_dev" type="str" value="$(arg i2c_dev)" /> 
    <param name="spi_dev" type="str" value="$(arg spi_dev)" />
    <param name="alsa_dev" type="str" value="$(arg alsa_dev)" />

    <param name="pga_gain" type="double" value="$(arg pga_gain)" />
    <param name="pga_offset" type="double" value="$(arg pga_offset)" />

    <param name="swcf_a_f0" type="double" value="$(arg swcf_a_f0)" />
    <param name="swcf_a_q" type="double" value="$(arg swcf_a_q)" />
    <param name="swcf_a_mode" type="int" value="$(arg swcf_a_mode)" />
    <param name="swcf_b_f0" type="double" value="$(arg swcf_b_f0)" />
    <param name="swcf_b_q" type="double" value="$(arg swcf_b_q)" />
    <param name="swcf_b_mode" type="int" value="$(arg swcf_b_mode)" />
</node>

<node pkg="nodelet" type="nodelet" name="evcam_node" args="standalone metavision_driver/DriverNodelet" clear_params="true" output="screen">
    <param name="use_multithreading" value="false"/>
    <param name="event_message_time_threshold" value="0.001"/>
    <param name="trigger_message_time_threshold" value="0"/>
    <param name="sync_mode" value="secondary"/>
    <param name="trigger_in_mode" value="external"/>
    <param name="statistics_print_interval" value="1.0"/>
    <param name="bias_file" value="$(find u2dar_receiver)/config/recording_2024-04-22_19-08-00.bias" />
    <rosparam command="load" file="$(find metavision_driver)/config/trigger_pins.yaml"/>
</node>

</group>

</launch>
